### PR TITLE
[JIT] throw on custom forward for module containers

### PR DESCRIPTION
**Pull request by eellison**
Custom forwards of containers would silently not be compiled previously. Throw an error now instead. 

Fix for https://github.com/pytorch/pytorch/issues/26671

### CODE REVIEW

File: torch/jit/_recursive.py [hunk 1]
```diff
@@ -223,6 +223,10 @@ def try_compile_fn(fn, loc):
     return torch.jit.script(fn, _rcb=rcb)
 
 
+def check_custom_container(module, type):
```
Comment (driazati) [inline torch/jit/_recursive.py]: nit: don't shadow `type`

**Review by driazati — APPROVED** [review]
We should eventually support custom `forward`s on these containers, can you open another issue for that?

**Review by suo — DISMISSED** [review]
can you implement this for ScriptModule inheritance as well?

**Comment by eellison** [comment]
> can you implement this for ScriptModule inheritance as well?

What is the status of ScriptModules going through the same recursive path?

**Comment by suo** [comment]
> What is the status of ScriptModules going through the same recursive path?

The stack is accepted, just need to rebase/land. So probably going in early next week.

**Review by suo — APPROVED** [review]

**Comment by suo** [comment]
derpaderp I landed my module changes earlier than I thought I would. Sorry for the churn

**Comment by suo** [comment]
Heads up @eellison that this broke master tests, so I reverted :(

**Pull request closed without merging**